crates/primitives/src/key_ptr.rs

```diff
@@ -18,7 +18,7 @@ use crate::Key;
 ///
 /// This wraps a base key and provides an interface to mimic pointer arithmetics.
 /// Mainly used to coordinate keys through static storage structures.
-#[derive(Debug, Copy, Clone)]
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
```

Comment:
If these trait impls are only required for testing we might want to put them under `cfg_attr(test, ..)`

Reply:
So it's a good idea that I hadn't thought about. But as it turns out when we run tests for crate `storage`, crate `primitives` (which contains the above `KeyPtr`) will not have `cfg_attr(test)` true.

There's discussion about an RFC [here](https://github.com/rust-lang/cargo/issues/2911#issuecomment-625731513).